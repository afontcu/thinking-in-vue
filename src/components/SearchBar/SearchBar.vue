<template>
  <form>
    <input class="input" type="text" placeholder="Search..." />
    <Checkbox name="filterOffers" label="Show only offers" />
  </form>
</template>

<script>
import Checkbox from '@/components/Checkbox'

export default {
  name: 'SearchBar',
  components: {
    Checkbox
  }
}
</script>

<style scoped>
form {
  display: flex;
  align-items: center;
  margin: 1rem 0 2rem;
}
.input {
  flex: 1;
  background-color: hsl(210, 9%, 96%);
  padding: 0.9rem;
  font-size: 1rem;
  margin-right: 1rem;
  border: 0;
}
</style>
