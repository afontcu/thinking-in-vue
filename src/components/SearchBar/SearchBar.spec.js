import SearchBar from './index.js'
import { mount } from '@vue/test-utils'

describe('SearchBar.vue', () => {
  it('is a component', () => {
    const wrapper = mount(SearchBar)
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('renders an snapshot', () => {
    const wrapper = mount(SearchBar)
    expect(wrapper.html()).toMatchSnapshot('default')
  })
})
