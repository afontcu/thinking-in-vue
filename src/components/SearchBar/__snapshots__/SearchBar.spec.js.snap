// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar.vue renders an snapshot: default 1`] = `
<form>
    <input placeholder="Search..." class="input">
    <p class="filterOffers">
        <input type="checkbox" id="filterOffers">
        <label for="filterOffers">Show only offers</label>
    </p>
</form>
`;
